### PR TITLE
added to_masked_array method and masked_array property 

### LINKED ISSUE #460: Create method to return a masked array
This is mainly for plotting in MPL, which does not like NaNs.

### PR DESCRIPTION
**Pull request by jhamman**
This PR adds convienience methods to convert DataArrays to numpy masked arrays.

`DataArray.to_masked_array(copy=True)` returns a numpy Masked array 

`DataArray.masked_array` returns a masked array that is a view to the `DataArray.values.`

closes #460

### CODE REVIEW

File: xray/core/dataarray.py [hunk 1]
```diff
@@ -333,6 +334,11 @@ def values(self):
         """The array's data as a numpy.ndarray"""
         return self.variable.values
 
+    @property
+    def masked_array(self):
```
Comment (clarkfitzg) [inline xray/core/dataarray.py]: Do we need this? I feel like the method `to_masked_array` is sufficient.

File: xray/core/dataarray.py [hunk 2]
```diff
@@ -912,6 +918,22 @@ def to_series(self):
         index = self.coords.to_index()
         return pd.Series(self.values.reshape(-1), index=index, name=self.name)
 
+    def to_masked_array(self, copy=True):
```
Comment (clarkfitzg) [inline xray/core/dataarray.py]: Could make this a bit more general with something like:
```
to_masked_array(self, mask=None, masknull=True, copy=True):
    if masknull and not mask:
           mask = pd.isnull(self.values)
    return np.ma.MaskedArray(darray.values, mask)
```

Comment (clarkfitzg) [inline xray/core/dataarray.py]: Actually should be `if masknull mask is None:`

File: xray/core/dataarray.py (hunk 1 above)
Comment (jhamman) [inline xray/core/dataarray.py]: I think I'm in agreement here.  The benefit is cleaner syntax when plotting but I don't think it is required. Will remove.

File: xray/core/dataarray.py (hunk 2 above)
Comment (jhamman) [inline xray/core/dataarray.py]: sure.  Out of curiosity, why `pd.isnull` over `np.isfinite`?

Comment (clarkfitzg) [inline xray/core/dataarray.py]: @shoyer showed me that `pd.isnull` can be more robust. One example-

```
In [24]: a = np.array([None, 1, 2])

In [25]: a
Out[25]: array([None, 1, 2], dtype=object)

In [26]: np.isfinite(a)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-26-65dd1c860e4b> in <module>()
----> 1 np.isfinite(a)

TypeError: ufunc 'isfinite' not supported for the input types, and the inputs could not be safely coerced to any supported types according to the casting rule ''safe''

In [27]: pd.isnull(a)
Out[27]: array([ True, False, False], dtype=bool)
```

File: xray/core/dataarray.py (hunk 1 above)
Comment (shoyer) [inline xray/core/dataarray.py]: Agreed

File: xray/core/dataarray.py (hunk 2 above)
Comment (shoyer) [inline xray/core/dataarray.py]: I don't see much value in the mask option or letting the user choose not to mask out nulls. If you want that, you can just use the MaskedArray constructor directly.

**Comment by jhamman** [comment]
503c0d0 removed the `masked_array` property and added the use of `pd.isnull` to calculate the mask.

File: xray/core/dataarray.py [hunk 3]
```diff
@@ -912,6 +913,23 @@ def to_series(self):
...
+        """Convert this array into a numpy.ma.MaskedArray
+
+        Parameters
+        ----------
+        copy : bool
+            If True (default) make a copy of the array in the result. If False
+            modify array in place and return a view.
```
Comment (shoyer) [inline xray/core/dataarray.py]: "Modify array in place" does not sound entirely accurate to me -- the array on the xray object is not modified

**Comment by shoyer** [comment]
Looks like you still need to update the docs?

**Comment by jhamman** [comment]
Done. I did need to update the docs.

File: xray/test/test_dataarray.py [hunk 4]
```diff
@@ -1324,6 +1324,41 @@ def test_to_and_from_series(self):
...
+        da = DataArray(x_masked)
+
+        # Test round trip
+        x_masked_2 = da.to_masked_array()
+        da_2 = DataArray(x_masked_2)
+        self.assertArrayEqual(x_masked, x_masked_2)
+        self.assertDataArrayEqual(da, da_2)
+
+        da_masked_array = da.to_masked_array(copy=True)
+        self.assertTrue(isinstance(da_masked_array, np.ma.MaskedArray))
```
Comment (shoyer) [inline xray/test/test_dataarray.py]: note: you can use `assertIsInstance` here

**Comment by shoyer** [comment]
@jhamman Looks good to me. Feel free to merge.

**Pull request merged**